test/test.sh

```diff
@@ -1,30 +1,58 @@
-#!/bin/bash -e
+#!/bin/sh
+
+set -eu
 
 export TZ='US/Pacific'
 
-for pcap in `ls test/pcap`
-do
+cleanup()
+{
+    if [ -d "${TMP}" ] && [ "${TMP#/tmp/tmp.x}" != "${TMP}x" ]; then
+        rm -rf "${TMP}"
+    fi
+}
+
+trap cleanup EXIT
+
+for pcap in test/pcap/*.pcap; do
     TMP=$(mktemp -d)
     if [ -z "${TMP}" ]; then
         echo "Failed to create temporary directory '${TMP}'"
         break
     fi
     TMPSE2CSV="${TMP}/se2csv"
 
-    SAMPLE="${pcap%.*}"
+    SAMPLE=$(basename --suffix=".pcap" "${pcap}")
     KEY_OPTION=""
     if [ -f "test/keys/${SAMPLE}.key" ]
     then
         KEY_OPTION="-k test/keys/${SAMPLE}.key"
     fi
     echo "Running test sample: ${SAMPLE}"
     echo "Key option: ${KEY_OPTION}"
-    tshark -r "test/pcap/${SAMPLE}.pcap" -T fields -e data | ./utilities/unhexlify.py | ./semonitor.py -r "${TMP}/${SAMPLE}.rec" -o "${TMP}/${SAMPLE}.json" - $KEY_OPTION
-    diff "${TMP}/${SAMPLE}.json" "test/json/${SAMPLE}.json"
-    cmp -l "${TMP}/${SAMPLE}.rec" "test/rec/${SAMPLE}.rec"
+    if ! tshark -r "test/pcap/${SAMPLE}.pcap" -T fields -e data | ./utilities/unhexlify.py | ./semonitor.py -r "${TMP}/${SAMPLE}.rec" -o "${TMP}/${SAMPLE}.json" - ${KEY_OPTION}; then
```

Comment:
why do you want to test the output of each command individually?  isn't this why we set -e?

Reply:
that way, we can print an error message indicating why things failed. set -eu will just exit, and in cases silently as well if the previous does not print why things fail. This just adds an extra layer of verboseness.

Reply:
how about adding -x to the set command so we'll know exactly the command that failed.  i'm not a fan of conditionally checking each and every command because it adds a lot of error-prone boilerplate

Reply:
sorry it took me this long to work on this again; but i cleaned it up as suggested and a few more sprinkles